avoid black screen(for the entire boot session) when starting X with radeon DRM 
when things are built as modules
by automatically loading fbcon module when DRM is selected (CONFIG_DRM) because
fbcon is a requirement (ie. can't have old text mode consoles and X, they must be framebuffer consoles before X is started) according to knowledgeable people on #radeon /freenode

diff --git a/drivers/gpu/drm/Kconfig b/drivers/gpu/drm/Kconfig
index 1a0a8df..da572fe 100644
--- a/drivers/gpu/drm/Kconfig
+++ b/drivers/gpu/drm/Kconfig
@@ -35,7 +35,7 @@ config DRM_KMS_FB_HELPER
 	bool
 	depends on DRM_KMS_HELPER
 	select FB
-	select FRAMEBUFFER_CONSOLE if !EXPERT
+	select FRAMEBUFFER_CONSOLE
 	select FRAMEBUFFER_CONSOLE_DETECT_PRIMARY if FRAMEBUFFER_CONSOLE
 	select FB_SYS_FOPS
 	select FB_SYS_FILLRECT
diff --git a/drivers/gpu/drm/drm_fb_helper.c b/drivers/gpu/drm/drm_fb_helper.c
index 418d299..e170478 100644
--- a/drivers/gpu/drm/drm_fb_helper.c
+++ b/drivers/gpu/drm/drm_fb_helper.c
@@ -2000,7 +2000,7 @@ EXPORT_SYMBOL(drm_fb_helper_hotplug_event);
  * but the module doesn't depend on any fb console symbols.  At least
  * attempt to load fbcon to avoid leaving the system without a usable console.
  */
-#if defined(CONFIG_FRAMEBUFFER_CONSOLE_MODULE) && !defined(CONFIG_EXPERT)
+#if defined(CONFIG_FRAMEBUFFER_CONSOLE_MODULE) && ( !defined(CONFIG_EXPERT) || defined(CONFIG_DRM_MODULE) || defined(CONFIG_DRM) )
 static int __init drm_fb_helper_modinit(void)
 {
 	const char *name = "fbcon";
